libr/core/cmd_anal.c

```diff
@@ -2035,19 +2036,83 @@ static int cmd_anal_fcn(RCore *core, const char *input) {
 			free (dbpath);
 			} break;
 		case'?':
-			r_core_cmd_help (core, help_afC);
+			r_core_cmd_help (core, help_afc);
 			break;
-		case 'l': //afCl list all function Calling conventions.
+		case 'l': //afcl list all function Calling conventions.
 			sdb_foreach (core->anal->sdb_cc, cc_print, NULL);
 			break;
 		case 'a':
 			eprintf ("Todo\n");
 			break;
+		case 'r': {	// afcr
+			int i;
+			char *out, *cmd, *regname, *tmp;
+			char *subvec_str = r_str_new ("");
+			char *json_str = r_str_new ("");
+			// if json_str initialize to NULL, it's possible for afcrj to output a (null)
+			// subvec_str and json_str should be valid until exiting this code block
+			bool json = input[3] == 'j'? true: false;
+			for (i = 0; i <= 11; i++) {
+				if (i == 0) {
+					cmd = r_str_newf ("cc.%s.ret", fcn->cc);
+				} else {
+					cmd = r_str_newf ("cc.%s.arg%d", fcn->cc, i);
+				}
+				if (i < 7) {
+					regname = r_str_new (cmd);
+				} else {
+					regname = r_str_newf ("cc.%s.float_arg%d", fcn->cc, i - 6);
+				}
+				out = sdb_querys (core->anal->sdb_cc, NULL, 0, cmd);
+				free (cmd);
+				if (out) {
+					out[strlen (out) - 1] = 0;
+					if (json) {
+						tmp = subvec_str;
+						subvec_str = r_str_newf ("%s,\"%s\"", subvec_str, out);
+						free (tmp);
+					} else {
+						r_cons_printf ("%s: %s\n", regname, out);
+					}
+					free (out);
+				}
+				free (regname);
+				if (!subvec_str[0]) {
+					continue;
+				}
+				switch (i) {
+				case 0: {
+					tmp = json_str;
+					json_str = r_str_newf ("%s,\"ret\":%s", json_str, subvec_str + 1);
```

Comment:
because I realise the three cases have the same status and want to reduce the similar code, but not sure if it's suitable as follows

```
char *format = NULL;
switch (i) {
case 0 : 
        format = r_str_new("%s,\"ret\":%s");
        break;
case 6 : 
        format = r_str_new("%s,\"args\":[%s]");
        break;
case 11 : 
        format = r_str_new("%s,\"float_args\":[%s]");
        break;
default:
        continue;
}
tmp = json_str;
json_str = r_str_newf (format, json_str, subvec_str + 1);
free (tmp);
```